pkg/sql/conn_executor_exec.go

```diff
@@ -3320,6 +3320,8 @@ func (ex *connExecutor) recordTransactionFinish(
 		)
 	}
 
+	ex.statsCollector.ObserveTransaction(ctx, transactionFingerprintID, recordedTxnStats)
```

Comment:
Similar to prior commit, do we need a conditional here to check if ID is 0?

Reply:
I'll remove the stmtFIngerprintId == 0 check in the statement recording. It's not necessary.